pkg/types/types.go

```diff
@@ -12,8 +12,10 @@ type Severity int
 type VendorSeverity map[string]Severity
 
 type CVSSVector struct {
-	V2 string `json:"v2,omitempty"`
-	V3 string `json:"v3,omitempty"`
+	V2      string  `json:"v2,omitempty"`
+	V2Score float64 `json:"v2_score,omitempty"`
+	V3      string  `json:"v3,omitempty"`
+	V3Score float64 `json:"v3_score,omitempty"`
```

Comment:
I feel score is not vector. How about
```suggestion
	V2      string  `json:"v2,omitempty"`
	V3      string  `json:"v3,omitempty"`
	
type CVSSScore struct {
	V2 float64 `json:"v2,omitempty"`
	V3 float64 `json:"v3,omitempty"`
}
```

Or

```
type CVSS struct {
	V2Score float64 `json:"v2,omitempty"`
	V3Score float64 `json:"v3,omitempty"`
	V2Vector      string  `json:"v2,omitempty"`
	V3Vector     string  `json:"v3,omitempty"`
}
```

Reply:
Got it. Let's go with the second approach. https://github.com/aquasecurity/trivy-db/pull/48/commits/53c090dc1bb65c875d16c54efea5b9aa6dd69aa0

Reply:
To keep changes backwards compatible, I've changed the approach to introduce a new struct instead (Approach #1) https://github.com/aquasecurity/trivy-db/pull/48/commits/9199281e1c1e51903efc19762830bbfacb240469

Reply:
How about the following with the first approach?


```
type Vulnerability struct {
	Title          string         `json:",omitempty"`
	Description    string         `json:",omitempty"`
	Severity       string         `json:",omitempty"`
	VendorSeverity VendorSeverity `json:",omitempty"`
	VendorScores  VendorScores  `json:",omitempty"`
	VendorVectors  VendorVectors  `json:",omitempty"`
	References     []string       `json:",omitempty"`
}
```

Or, 

```
type Vulnerability struct {
	Title          string         `json:",omitempty"`
	Description    string         `json:",omitempty"`
	Severity       string         `json:",omitempty"`
	VendorSeverity VendorSeverity `json:",omitempty"`
	VendorVectors  VendorVectors  `json:",omitempty"` // only for backward compatibility
	VendorCVSSs  VendorVectors  `json:",omitempty"`
	References     []string       `json:",omitempty"`
}
```

with the second approach.

To be honest, I feel this is a bit weird 😁

```
type CVSSVector struct {
	V2     string `json:"v2,omitempty"`
	V3     string `json:"v3,omitempty"`
	Scores CVSSScore
}
```

Reply:
OK, what do you want `VendorScores` to be of type then? Is it a `map[string]CVSSScore` where `CVSSScore` is `type CVSSScore struct { V2 float64, V3 float64 }` or something else?

If it's the case we'll have to add additional logic to `vulnerability.go` to add a new method called `getVendorScores` which will be called by `GetDetail`. I feel it's unnecessary.

Maybe we could rename like the following, if that works for you:

```
type CVSSVectorsAndScores struct {
	V2     string `json:"v2,omitempty"`
	V3     string `json:"v3,omitempty"`
	Scores CVSSScore
}
```

But I feel this discussion doesn't add much technical value at this point.

Reply:
Hmm...It looks different from my suggestion... It seems to break compatibility because the same key `VendorVectors` uses a different type `VendorCVSS`. I think an old trivy can't decode a new JSON and returns an error.

Reply:
> Hmm...It looks different from my suggestion... It seems to break compatibility because the same key `VendorVectors` uses a different type `VendorCVSS`. I think an old trivy can't decode a new JSON and returns an error.

Sorry I misread your earlier comment. I addressed it here again: https://github.com/aquasecurity/trivy-db/pull/48/commits/febb18e734ec752faa9d9d4773af3af4e784c7a0


I just tried with the master version of Trivy, it works fine (no errors) but it doesn't show any VendorVector information (CVSS Vectors) after this new commit. This is as expected because now we store `CVSS` field instead of `VendorVector` so old Trivy doesn't know about it. 

I addressed it here https://github.com/aquasecurity/trivy-db/pull/48/commits/2e0299ee4f62afdd7384a474618aac79b045f338 now old Trivy can use new DB and display `VendorVectors` and new Trivy can display `CVSS` which includes both.